pkg/vsphere/vm/vm.go

```diff
@@ -272,17 +272,27 @@ func (vm *VirtualMachine) DeleteExceptDisks(ctx context.Context) (*types.TaskInf
 
 	op := trace.FromContext(ctx, "DeleteExceptDisks")
 
-	op.Debugf("Getting list of the devices for VM %q", vm)
-	devices, err := vm.Device(op)
-	if err != nil {
-		return nil, err
-	}
+	firstTime := true
+	err := retry.Do(func() error {
+		op.Debugf("Getting list of the devices for VM %q", vm)
```

Comment:
We may want to use vm.ObjectName() instead of just vm as it stands. Currently we will be logging the moref of the VM which is not what we use in any of the other log locations.